src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -802,6 +802,62 @@ static double validateAndParseReal(String columnName, Object input, long insertR
         columnName, input.getClass(), "REAL", new String[] {"Number", "String"}, insertRowIndex);
   }
 
+  /**
+   * Validates and parses input Iceberg INT column. Allowed Java types:
+   *
+   * <ul>
+   *   <li>Number
+   *   <li>String
+   * </ul>
+   *
+   * @param columnName Column name, used in validation error messages
+   * @param input Object to validate and parse
+   * @param insertRowIndex Row index for error reporting
+   * @return Parsed integer
+   */
+  static int validateAndParseIcebergInt(String columnName, Object input, long insertRowIndex) {
+    if (input instanceof Number) {
+      return ((Number) input).intValue();
+    } else if (input instanceof String) {
+      try {
+        return Integer.parseInt(((String) input).trim());
+      } catch (NumberFormatException e) {
+        throw valueFormatNotAllowedException(
+            columnName, "INT", "Not a valid integer", insertRowIndex);
+      }
+    }
+    throw typeNotAllowedException(
+        columnName, input.getClass(), "INT", new String[] {"Number", "String"}, insertRowIndex);
+  }
+
+  /**
+   * Validates and parses input Iceberg LONG column. Allowed Java types:
+   *
+   * <ul>
+   *   <li>Number
+   *   <li>String
+   * </ul>
+   *
+   * @param columnName Column name, used in validation error messages
+   * @param input Object to validate and parse
+   * @param insertRowIndex Row index for error reporting
+   * @return Parsed long
+   */
+  static long validateAndParseIcebergLong(String columnName, Object input, long insertRowIndex) {
+    if (input instanceof Number) {
+      return ((Number) input).longValue();
```

Comment:
when all can .longValue throw exceptions?

Reply:
Discussed via slack. Added range check and rounding.